src/bin/example-read.rs

```diff
@@ -0,0 +1,151 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+extern crate parquet;
+
+use std::fs::File;
+use std::path::Path;
+
+use parquet::basic::*;
+use parquet::data_type::*;
+use parquet::column::reader::{ColumnReaderImpl, get_typed_column_reader};
+use parquet::file::reader::{FileReader, SerializedFileReader};
+
+// Run example
+// ===
+// $ cargo run --bin example-read
+//
+// Console output
+// ===
+// Reading batch of records: 4
+// Record { b: 2, d_c: "abc", e: [1, 1] }
+// Record { b: 3, d_c: "def", e: [10, 10] }
+// Record { b: 4, d_c: "ghi", e: [100, 100] }
+// Record { b: 4, d_c: "jkl", e: [1000, 1000] }
+//
+
+#[derive(Clone, Debug, Default)]
+struct Record { b: i64, d_c: String, e: Vec<i32> }
+
+// read values from column reader for requested batch size, can read fewer records
+fn read_values<T: DataType>(
+    reader: &mut ColumnReaderImpl<T>,
+    batch_size: usize) -> (usize, Vec<T::T>, Vec<i16>, Vec<i16>) where T: 'static {
+
+  let mut values = vec![T::T::default(); batch_size];
+  let mut def_levels = vec![i16::default(); batch_size];
+  let mut rep_levels = vec![i16::default(); batch_size];
+  let mut curr_values_read = 0;
+  let mut curr_levels_read = 0;
+  loop {
```

Comment:
This loop doesn't seem to work if number of values is greater than `batch_size`.

Reply:
Fixed, thanks!
Now I pass total values to read. Also when changing batch_size in `main()`, this will read and parse data accordingly.